File: schemaorg-rs/src/energy_consumption_details.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type EnergyConsumptionDetailsAdditionalTypeFieldEnum = String;
///<https://schema.org/EnergyConsumptionDetails>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct EnergyConsumptionDetails {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/hasEnergyEfficiencyCategory>
    #[serde(rename = "hasEnergyEfficiencyCategory")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_energy_efficiency_category: Vec<EnergyEfficiencyEnumeration>,
    ///<https://schema.org/energyEfficiencyScaleMin>
    #[serde(rename = "energyEfficiencyScaleMin")]
    #[serde_as(as = "OneOrMany<_>")]
    pub energy_efficiency_scale_min: Vec<EUEnergyEfficiencyEnumerationEnum>,
    ///<https://schema.org/energyEfficiencyScaleMax>
    #[serde(rename = "energyEfficiencyScaleMax")]
    #[serde_as(as = "OneOrMany<_>")]
    pub energy_efficiency_scale_max: Vec<EUEnergyEfficiencyEnumerationEnum>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<EnergyConsumptionDetailsMainEntityOfPageFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<EnergyConsumptionDetailsIdentifierFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<EnergyConsumptionDetailsDescriptionFieldEnum>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<EnergyConsumptionDetailsSubjectOfFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<EnergyConsumptionDetailsAdditionalTypeFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<EnergyConsumptionDetailsImageFieldEnum>,
}

File: schemaorg-ts/src/energy_consumption_details.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type EnergyConsumptionDetailsAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/EnergyConsumptionDetails>
 **/
export interface EnergyConsumptionDetails {
  "@context": string;
  "@type": "EnergyConsumptionDetails";

  /**
   * <https://schema.org/hasEnergyEfficiencyCategory>
   **/
  hasEnergyEfficiencyCategory: all.OneOrMany<all.EnergyEfficiencyEnumeration>;

  /**
   * <https://schema.org/energyEfficiencyScaleMin>
   **/
  energyEfficiencyScaleMin: all.OneOrMany<all.EUEnergyEfficiencyEnumerationEnum>;

  /**
   * <https://schema.org/energyEfficiencyScaleMax>
   **/
  energyEfficiencyScaleMax: all.OneOrMany<all.EUEnergyEfficiencyEnumerationEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.EnergyConsumptionDetailsMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.EnergyConsumptionDetailsIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.EnergyConsumptionDetailsDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.EnergyConsumptionDetailsSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.EnergyConsumptionDetailsAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.EnergyConsumptionDetailsImageFieldEnum>;
}
